### PR TITLE
Add aws s3:// uri support

### PR DESCRIPTION
**Pull request by tmnuwan12**
Fixes gh-361

### CODE REVIEW

File: spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/io/s3/SimpleStorageResource.java [hunk 1]
```diff
@@ -143,6 +145,10 @@ public URL getURL() throws IOException {
 				"/" + this.bucketName + "/" + this.objectName);
 	}
 
+	public URI getS3URI() throws URISyntaxException {
```
Comment (maciejwalkowiak) [inline spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/io/s3/SimpleStorageResource.java:148]: Lets change method name to `getS3Uri` - when you look at other classes in Spring Framework whenever they have `URI` in name its spelled `Uri`.

`URISyntaxException` is rather impossible to happen right? We should rather catch it and throw `new RuntimeException` with explanation, similar to how it's done in `AmazonS3ClientFactory`.

File: spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/io/s3/SimpleStorageResource.java [hunk 2]
```diff
@@ -143,6 +145,10 @@ public URL getURL() throws IOException {
...
+		return new URI("S3", "//" + this.bucketName + "/" + this.objectName, null);
```
Comment (maciejwalkowiak) [inline spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/io/s3/SimpleStorageResource.java:149]: `S3` should be rather lowercase `s3`

File: spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/io/s3/SimpleStorageResource.java (hunk 1 above)
Comment (tmnuwan12) [reply spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/io/s3/SimpleStorageResource.java]: Ok. Fixed as requested.

File: spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/io/s3/SimpleStorageResource.java (hunk 2 above)
Comment (tmnuwan12) [reply spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/io/s3/SimpleStorageResource.java]: Ok. Fixed as requested.

**Comment by tmnuwan12** [comment]
> Compilation issues is a result of Spring Boot just recently dropping JUnit 4. I'll add very soon JUnit 4 to dependencies both on `master` and `2.2.x` branches so make sure to rebase.

I have rebased against master.

**Comment by maciejwalkowiak** [comment]
Thanks @tmnuwan12! Your contribution has been merged.

**Pull request closed without merging**